src/main/java/com/dotloop/loopit/LoopItController.java

```diff
@@ -54,51 +51,102 @@
     @Value("${dotloop.api.endpoint}")
     private String apiEndpoint;
 
-    @RequestMapping("/loopit")
+    private String username = "api user";
+
+    private Request attachHeaders(Request request) {
+        Token token = TokenStore.get(username);
+        request.addHeader("Authorization", "Bearer " + token.getAccessToken())
+                .addHeader("Content-type", "application/json");
+
+        return request;
+    }
+
+    private Request attachBody(Request request,  String bodyData) {
+        request.bodyString(bodyData, ContentType.APPLICATION_JSON);
+
+        return request;
+    }
+
+    private String makeRequest(String type, String params, String bodyData) throws Exception {
+        try {
+            params = params.isEmpty() ? "" : params;
+
+            Request request = attachHeaders(Request.Get(getApiBaseUrl() + params));
+
+            if (type.equalsIgnoreCase("post")) {
+                request = attachBody(attachHeaders(Request.Post(getApiBaseUrl() + params)), bodyData);
```

Comment:
@sweetcoco this is a little hard to parse, ideally we'd have an object and then call a series of actions on it , where as this is the inverse of that pattern.  If you use the suggestion of creating dedicated request type methods, you can just attach the body directly to the Request object (`attachBody` is basically 1:1 swap code) and then set your headers in the shared request processing code.

Reply:
@sweetcoco Here's some (untested) code to give you an idea of what I was looking for:

```
    private String post(String url, String bodyData){

        Request postRequest = Request.Post(getApiBaseUrl() + url).bodyString(bodyData, ContentType.APPLICATION_JSON);
        return executeRequest(postRequest);
        
    }

    private String executeRequest(Request request){

        try {

            Token token = TokenStore.get(username);
            request.addHeader("Authorization", "Bearer " + token.getAccessToken())
                    .addHeader("Content-type", "application/json");


            HttpResponse response = request.execute().returnResponse();

            String payload = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
            logger.debug("Response: " + payload);
            if (response.getStatusLine().getStatusCode() >= 300) {
                throw new HttpResponseException(response.getStatusLine().getStatusCode(),
                        response.getStatusLine().getReasonPhrase());
            }

            return payload;

        } catch (Exception e) {
            logger.error("Something unexpected happened: " + e.getMessage());
            TokenStore.delete(username); // fixme - needed only for 401
            throw new RuntimeException(e); // todo handle error - token revoked, etc
        }

    }
```

Reply:
@adunn8 you really went above and beyond with this comment, thanks man.